src/kaggle.jl

```diff
@@ -27,7 +29,18 @@ function gen_kaggle_url(dataset)
     return "https://www.kaggle.com/api/v1/datasets/download/$dataset"
 end
 
-function kdownload(dataset, localdir)
+is_kaggle_url(url) = contains(url, KAGGLE_DOMAIN)
+
+function kaggle_url2dataset(url)
+    user_name = match(Regex("$KAGGLE_DOMAIN/([^\\/]+)/"), url).captures[]
+    dataset_name = match(Regex("$user_name/([^\\/]+)"), url).captures[]
+
+    return "$user_name/$dataset_name"
+end
+
+function kdownload(url_or_dataset, localdir)
+    dataset = is_kaggle_url(url_or_dataset) ? kaggle_url2dataset(url_or_dataset) : url_or_dataset
```

Comment:
Do we need some error handling if receiving ill-from or non-kaggle url?

Reply:
Maybe we do need, but this may be difficult to distinguish if we support both `dataset` and `url` pattern.

Reply:
For dataset pattern, it should be two legal name separate by `/` (definition of legal name should be found on kaggle's webpage).

And for url pattern, I believe it should start with `https` (or `http`).